src/compiler/scala/tools/nsc/transform/Fields.scala

```diff
@@ -412,58 +482,127 @@ abstract class Fields extends InfoTransform with ast.TreeDSL with TypingTransfor
     if (!module.isStatic) module setFlag METHOD | STABLE
   }
 
-  class FieldsTransformer(unit: CompilationUnit) extends TypingTransformer(unit) {
-    def mkTypedUnit(pos: Position) = localTyper.typedPos(pos)(CODE.UNIT)
+  class FieldsTransformer(unit: CompilationUnit) extends TypingTransformer(unit) with CheckedAccessorSynthTransformation {
+    protected def typedPos(pos: Position)(tree: Tree): Tree = localTyper.typedPos(pos)(tree)
+
+    def mkTypedUnit(pos: Position) = typedPos(pos)(CODE.UNIT)
+    // TODO: clean up. this method is not used
     def deriveUnitDef(stat: Tree)  = deriveDefDef(stat)(_ => mkTypedUnit(stat.pos))
 
-    def mkAccessor(accessor: Symbol)(body: Tree) = localTyper.typedPos(accessor.pos)(DefDef(accessor, body)).asInstanceOf[DefDef]
+    def mkAccessor(accessor: Symbol)(body: Tree) = typedPos(accessor.pos)(DefDef(accessor, body)).asInstanceOf[DefDef]
+
+    // this makes trees for mixed in fields, as well as for bitmap fields (their RHS will be EmptyTree because they are initialized implicitly)
+    // if we decide to explicitly initialize, use this RHS: if (symbol.info.typeSymbol.asClass == BooleanClass) FALSE else ZERO)
+    // could detect it's a bitmap field with something like `sym.name.startsWith(nme.BITMAP_PREFIX)` (or perhaps something more robust...)
+    def mkField(sym: Symbol, rhs: Tree = EmptyTree) = typedPos(sym.pos)(ValDef(sym, rhs)).asInstanceOf[ValDef]
+
+    /**
+      * Desugar a local `lazy val x: Int = rhs` into
+      * ```
+      * val x$lzy = new scala.runtime.LazyInt()
+      * def x(): Int =
+      *   x$lzy.synchronized {
+      *     if (!x$lzy.initialized) {
+      *       x$lzy.initialized = true
+      *       x$lzy.value = rhs
+      *     }
+      *     x$lzy.value
+      *  }
+      * ```
+      */
+    private def mkLazyLocalDef(lazyVal: Symbol, rhs: Tree): Tree = {
+      val lazyValType = lazyVal.tpe.resultType
+      val refClass = lazyHolders.getOrElse(lazyValType.typeSymbol, LazyRefClass)
+      val refTpe = if (refClass != LazyRefClass) refClass.tpe else appliedType(refClass.typeConstructor, List(lazyValType))
+
+      val flags = (lazyVal.flags & FieldFlags | ARTIFACT | MUTABLE) & ~(IMPLICIT | STABLE) // TODO: why include MUTABLE???
```

Comment:
Getting rid of `MUTABLE` here avoids an unnecessary `ObjectRef`.

Before:

```scala
⚡ qscalac -Xprint:mixin $(f "class C { def foo = { lazy val x = 42; x }}")
[[syntax trees at end of                     mixin]] // a.scala
package <empty> {
  class C extends Object {
    def foo(): Int = {
      lazy <artifact> var x$lzy: runtime.ObjectRef = scala.runtime.ObjectRef.create(new scala.runtime.LazyInt());
      C.this.x$1(x$lzy)
    };
    final private[this] def x$1(x$lzy$1: runtime.ObjectRef): Int = scala.Int.unbox(x$lzy$1.elem.$asInstanceOf[scala.runtime.LazyInt]().synchronized({
      if (x$lzy$1.elem.$asInstanceOf[scala.runtime.LazyInt]().initialized().unary_!())
        {
          x$lzy$1.elem.$asInstanceOf[scala.runtime.LazyInt]().initialized_=(true);
          x$lzy$1.elem.$asInstanceOf[scala.runtime.LazyInt]().value_=(42)
        };
      scala.Int.box(x$lzy$1.elem.$asInstanceOf[scala.runtime.LazyInt]().value())
    }));
    def <init>(): C = {
      C.super.<init>();
      ()
    }
  }
}
```

After:

```scala
[[syntax trees at end of                     mixin]] // a.scala
package <empty> {
  class C extends Object {
    def foo(): Int = {
      lazy <artifact> val x$lzy: scala.runtime.LazyInt = new scala.runtime.LazyInt();
      C.this.x$1(x$lzy)
    };
    final private[this] def x$1(x$lzy$1: scala.runtime.LazyInt): Int = scala.Int.unbox(x$lzy$1.synchronized({
      if (x$lzy$1.initialized().unary_!())
        {
          x$lzy$1.initialized_=(true);
          x$lzy$1.value_=(42)
        };
      scala.Int.box(x$lzy$1.value())
    }));
    def <init>(): C = {
      C.super.<init>();
      ()
    }
  }
}
```